File: backend/src/models/Task.js
const mongoose = require('mongoose');

const TimeLogSchema = new mongoose.Schema({
  startedAt: { type: Date, required: true },
  endedAt:   { type: Date },
  duration:  { type: Number, default: 0 },
});

const CommentSchema = new mongoose.Schema({
  text:        { type: String, required: true },
  author:      { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  isRejection: { type: Boolean, default: false },
}, { timestamps: true });

const TaskSchema = new mongoose.Schema({
  title:        { type: String, required: true },
  description:  { type: String, default: '' },
  type:         { type: String, enum: ['bug', 'feature', 'enhancement'], required: true },
  status:       { type: String, enum: ['todo', 'in-progress', 'in-qa', 'done'], default: 'todo' },
  project:      { type: mongoose.Schema.Types.ObjectId, ref: 'Project', required: true },
  createdBy:    { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  assignedTo:   { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  qaAssignedTo: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  queuePosition:{ type: Number, default: 0 },
  timeLogs:     [TimeLogSchema],
  totalTime:    { type: Number, default: 0 }, // persisted seconds
  comments:     [CommentSchema],
  rejectionCount: { type: Number, default: 0 },
}, { timestamps: true });

module.exports = mongoose.model('Task', TaskSchema);

File: backend/src/main/java/com/taskflow/task/domain/Task.java
package com.taskflow.task.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "tasks")
public class Task {

    @Id
    private String id;

    private String title;
    private String description;
    private String type; // bug | feature | enhancement

    @Builder.Default
    private TaskStatus status = TaskStatus.TODO;

    private String project; // Project id
    private String createdBy; // User id
    private String assignedTo; // User id, nullable
    private String qaAssignedTo; // User id, nullable

    @Builder.Default
    private List<TimeLog> timeLogs = new ArrayList<>();

    @Builder.Default
    private List<Comment> comments = new ArrayList<>();

    @Builder.Default
    private long totalTime = 0; // seconds

    @Builder.Default
    private int rejectionCount = 0;

    private int queuePosition;

    @Builder.Default
    private Instant createdAt = Instant.now();
}
